cs251tk/toolkit/__main__.py

```diff
@@ -172,11 +174,30 @@ def main():
             no_update=no_update,
             specs=specs,
             skip_web_compile=skip_web_compile,
-            stogit_url=stogit_url,
-            web=web
+            stogit_url=stogit_url
         )
 
-        if workers > 1:
+        if web:
+            Thread(target=run_server, args=(basedir,), daemon=True).start()
+            for user in usernames:
+                clone_student(user, baseurl=stogit_url)
+            do_record = launch_cli(basedir=basedir,
+                                   date=date,
+                                   no_update=no_update,
+                                   spec=specs[next(iter(assignments))],
```

Comment:
Sorry, I just noticed this one - I forget what type of data structure "assignments" is - can we avoid manually running the iterator  here? Or is this the cleanest method?

Reply:
`assignments` is a List of str: https://github.com/StoDevX/cs251-toolkit/pull/120/files/21009682da10a1ebc727e8ed51c87083f79b0bae#diff-b574e18ce1af2e2b58c8a46f01f5c8aaR138. That's the method I found worked, but there likely is a better way.

Reply:
When I try `spec=specs[assignments[0]]`, it gives me `TypeError: 'set' object does not support indexing`

Reply:
So if it's a set, it's inherently unordered - do you care which is the "first"?

If no, I'd do `list(assignments)[0]`; if yes, I'd do `sorted(assignments)[0]`.

Reply:
It should only contain one assignment at that point (because of the check in args.py), so `list(assignments)[0]` will work.